src/LinSys/linsysmerger.ci

```diff
@@ -200,7 +200,7 @@ module linsysmerger {
       entry void wait4lhs() {
         when lhsbc_complete() serial "hyprelhs" { hyprelhs(); } };
       entry void wait4rhs() {
-        when rhsbc_complete() serial "hyprerhs" { hyprerhs(); } };
+        when rhsbc_complete_rhs() serial "hyprerhs" { hyprerhs(); } };
```

Comment:
This is very good. I did not even thought about naming these differently, which does make the client code more readable. Thanks for this, this is awesome! There might be other duplicates too.

BTW: The reason behind these duplicate signals is my (most likely limited) understanding is that these behave like messages in a queue so when they can be consumed by multiple `when` SDAG statements, there must be multiple "messages" in the queue. If not, the incorrect asynchronous logic happens which can lead to errors, incorrect results, and stalling. More info about this is in the commit message of 5663ed146b2d12bac1654ca5fc53b4767e003622, in the paragraph starting with _This commit also revamps the asynchronous logic in LinSysMerger_. In particular, please see also my discussion on the Charm++ mailing list on this topic at https://lists.cs.illinois.edu/lists/arc/charm/2016-09/msg00002.html. The correct way of using SDAG triggers in a multiple fashion is certainly still a little bit still unclear to me.

In summary, the change does make the code cleaner, and I think the asynchronous logic stays correct. I also welcome any more insight on how to understand (think about) this issue better.